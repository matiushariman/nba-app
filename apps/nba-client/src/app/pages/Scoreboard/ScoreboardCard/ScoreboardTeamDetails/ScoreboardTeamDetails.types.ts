import type { Team } from '@nba-app/types-nba';

export type ScoreboardTeamDetailsProps = Pick<
  Team,
  'teamName' | 'teamId' | 'wins' | 'losses'
>;
