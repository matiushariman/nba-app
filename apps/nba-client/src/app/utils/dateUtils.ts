import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
