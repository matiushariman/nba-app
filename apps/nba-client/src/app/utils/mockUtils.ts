export const shouldUseMock = process.env['NODE_ENV'] === 'development';
