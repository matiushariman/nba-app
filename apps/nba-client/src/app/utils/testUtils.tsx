import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
