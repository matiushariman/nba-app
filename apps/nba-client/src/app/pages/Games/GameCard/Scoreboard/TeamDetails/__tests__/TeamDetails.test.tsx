import ScoreboardTeamDetails from '..';
import { render } from '../../../../../../utils/testUtils';

describe('pages/games/GameCard/Scoreboard/GameCard/TeamDetails', () => {
  it('should render TeamDetails that matches snapshot', () => {
    const { baseElement } = render(
      <ScoreboardTeamDetails
        teamId={123}
        teamName="Celtics"
        wins={82}
        losses={0}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
