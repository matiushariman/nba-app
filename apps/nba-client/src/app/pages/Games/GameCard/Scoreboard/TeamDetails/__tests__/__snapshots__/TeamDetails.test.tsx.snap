// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/games/GameCard/Scoreboard/GameCard/TeamDetails should render TeamDetails that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="grid gap-1 justify-center"
    >
      <img
        alt="Celtics logo"
        height="64"
        loading="lazy"
        src="https://cdn.nba.com/logos/nba/123/global/L/logo.svg"
        width="64"
      />
      <div>
        <p
          class="text-center text-sm"
        >
          Celtics
        </p>
        <p
          class="text-center text-gray-500 text-xs"
        >
          82 - 0
        </p>
      </div>
    </div>
  </div>
</body>
`;
